crates/burn-import/src/pytorch/reader.rs

```diff
@@ -48,7 +49,29 @@ where
         .collect();
 
     // Remap the keys (replace the keys in the map with the new keys)
-    let tensors = remap(tensors, key_remap);
+    let (tensors, remapped_keys) = remap(tensors, key_remap);
```

Comment:
I do not think we can zip the tensor keys and remapped because the order is not maintained. If I do not drain while iterating, we double memory. I could re-insert modified keys but I cannot delete while iterating. I think for now, it's the cleanest solution for now.

Alternatively we can use BTreeMap to hold tensors but it's not O(1) lookup and does not have features like .drain().

Reply:
Ahh ok makes sense! Let's keep your proposed solution then.